lib/internal/Magento/Framework/View/Test/Unit/Asset/RepositoryTest.php

```diff
@@ -157,36 +157,218 @@ public function testUpdateDesignParamsWrongTheme()
+        $this->assertSame($themeMock, $params['themeModel']);
     }
 
     /**
-     * @return array
+     * @return void
      */
-    public function updateDesignParamsDataProvider()
+    public function testUpdateDesignParamsCrossAreaThemeId()
     {
-        return [
-            [
-                ['area' => 'AREA'],
-                ['area' => 'AREA', 'themeModel' => '', 'module' => '', 'locale' => '']],
-            [
-                ['themeId' => 'ThemeID'],
-                ['area' => '', 'themeId' => 'ThemeID', 'themeModel' => 'ThemeID', 'module' => '', 'locale' => '']
-            ]
-        ];
+        $params = ['area' => 'frontend'];
+
+        $this->designMock
+            ->expects($this->any())
+            ->method('getDesignParams')
+            ->willReturn(
+                [
+                    'themeModel' => '',
+                    'area' => 'adminhtml',
+                    'locale' => ''
+                ]
+            );
+        $this->designMock
+            ->expects($this->any())
+            ->method('getConfigurationDesignTheme')
+            ->willReturn('1');
+
+        $themeMock = $this->getMock(\Magento\Framework\View\Design\ThemeInterface::class);
+        $themeMock->expects($this->any())
+            ->method('getArea')
+            ->willReturn('frontend');
+
+        $this->themeProvider
+            ->expects($this->any())
+            ->method('getThemeById')
+            ->with('1')
+            ->willReturn($themeMock);
+        $this->themeProvider
+            ->expects($this->never())
+            ->method('getThemeByFullPath');
+
+        $this->repository->updateDesignParams($params);
+        $this->assertEquals(
+            ['area' => 'frontend', 'themeModel' => $themeMock, 'module' => '', 'locale' => ''],
+            $params
+        );
+        $this->assertSame($themeMock, $params['themeModel']);
+    }
+
+    /**
+     * @return void
+     */
+    public function testUpdateDesignParamsCrossAreaThemePath()
+    {
+        $params = ['area' => 'frontend'];
+
+        $this->designMock
+            ->expects($this->any())
+            ->method('getDesignParams')
+            ->willReturn(
+                [
+                    'themeModel' => '',
+                    'area' => 'adminhtml',
+                    'locale' => ''
+                ]
+            );
+        $this->designMock
+            ->expects($this->any())
+            ->method('getConfigurationDesignTheme')
+            ->willReturn('Magento/luma');
+
+        $themeMock = $this->getMock(\Magento\Framework\View\Design\ThemeInterface::class);
+        $themeMock->expects($this->any())
+            ->method('getArea')
+            ->willReturn('frontend');
+
+        $this->themeProvider
+            ->expects($this->never())
+            ->method('getThemeById');
+        $this->themeProvider
+            ->expects($this->any())
+            ->method('getThemeByFullPath')
+            ->with('frontend/Magento/luma')
```

Comment:
Please avoid Luma theme mentioning in framework, just use some fictional theme path (fix both occurrences).